### PR TITLE
ripsaw | Getting the UUID of a test

### PR DESCRIPTION
**Pull request by Avilir**

### CODE REVIEW

**Comment by Avilir** [comment]
Verifyed locally, Fromthe log file :
11:28:54 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n my-ripsaw --kubeconfig /home/avili/Testing/Clusters/AWSCL/auth/kubeconfig exec fio-client-9156c74c-gtlkb -- env
11:28:56 - MainThread - ocs_ci.ocs.ripsaw - INFO - The UUID of the test is : 9156c74c-e833-5640-aa02-a2d55d9dc434
1

**Comment by wusui** [comment]
Looks good

File: ocs_ci/ocs/ripsaw.py [hunk 1]
```diff
@@ -88,6 +88,26 @@ def apply_crd(self, crd):
...
+        Return:
+            str: the UUID of the test
+        """
+        output = self.pod_obj.exec_oc_cmd(f'exec {benchmark} -- env')
+        uuid = ''
+        for line in output.split():
+            if 'uuid=' in line:
+                uuid = line.split('=')[1]
+        log.info(f'The UUID of the test is : {uuid}')
+        return uuid
```
Comment (clacroix12) [inline ocs_ci/ocs/ripsaw.py:109]: Should these lines be within the `if` statement or do we really want to keep iterating over the output after we have found the `uuid`? If these lines move into the `if` statement we can also remove `uuid = ''` from line 104

File: ocs_ci/ocs/ripsaw.py [hunk 2]
```diff
@@ -88,6 +88,26 @@ def apply_crd(self, crd):
...
+        """
+        Getting the UUID of the test.
+           when ripsaw used for running a benchmark tests, each run get its own
+           UUID, so the results in the elastic-search server can be sorted.
+
+        Args:
+            benchmark (str): the name of the main pod in the test
+
+        Return:
+            str: the UUID of the test
```
Comment (clacroix12) [inline ocs_ci/ocs/ripsaw.py:101]: Add an empty line before the end of the docstring

```suggestion
            str: the UUID of the test

```

File: ocs_ci/ocs/ripsaw.py (hunk 1 above)
Comment (Avilir) [reply ocs_ci/ocs/ripsaw.py:109]: I used this approach to make sure that the uuid is defined by the ripsaw benchmark operator.
If in future benchmarks run by the ripsaw will have a bug and this variable will not define, this test will not failed.

File: ocs_ci/ocs/ripsaw.py (hunk 2 above)
Comment (Avilir) [reply ocs_ci/ocs/ripsaw.py:101]: add

File: ocs_ci/ocs/ripsaw.py [hunk 3]
```diff
@@ -88,6 +88,27 @@ def apply_crd(self, crd):
...
+
+        """
+        output = self.pod_obj.exec_oc_cmd(f'exec {benchmark} -- env')
+        uuid = ''
+        for line in output.split():
+            if 'uuid=' in line:
+                uuid = line.split('=')[1]
```
Comment (ebenahar) [inline ocs_ci/ocs/ripsaw.py:108]: add `break` below

Comment (Avilir) [reply ocs_ci/ocs/ripsaw.py:108]: done

**Pull request merged**